src/transformers/modeling_openai.py

```diff
@@ -443,9 +444,7 @@ def forward(
 
         if position_ids is None:
             # Code is different from when we had a single embedding matrice from position and token embeddings
-            device = input_ids.device if input_ids is not None else inputs_embeds.device
-            position_ids = torch.arange(input_shape[-1], dtype=torch.long, device=device)
-            position_ids = position_ids.unsqueeze(0).view(-1, input_shape[-1])
+            position_ids = self.position_ids[None, : input_shape[-1]]
```

Comment:
Does this need to happen in other places? I'm confused why only openaigpt is failing.

Reply:
Yeah, I'd also be interested if this only happens for openai or other models as well...

Reply:
That's possible for some other models the same kind of issue. We need to discuss what kind of level of support we want to give to TorchScript, in such case we might need to deploy a broader unittests set as we do for ONNX

Reply:
@sshleifer I think we're not actively testing export to TorchScript so we're not monitoring issue(s) on this aspect.

Reply:
Should we make changes to these files as well?

[modeling_bert.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_bert.py#L191-L192)
[modeling_mobilebert.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_mobilebert.py#L189-L191)
[modeling_xlm.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_xlm.py#L517-L519)

Disclaimer: I did a cursory code match, so please ignore if this doesn't apply.

Reply:
I can address all of them on this PR @patrickvonplaten @sshleifer

Reply:
we also hit the same problem in gpt_2: https://github.com/huggingface/transformers/issues/5664#issuecomment-675511529

Reply:
if this indeed fixed the tracing issues, then the fix should be same for modeling_gpt2.py